File: leet_1492_kth_factor.cc
class Solution {
public:
    int kthFactor(int n, int k)
    {
        int i = 0;
        while (++i < n + 1)
        {
            if (n % i)
                continue ;
            --k;
            if (!k)
                return i;
        }
        return -1;
    }
};

File: leet_1492_kth_factor.go
func kthFactor(n int, k int) int {
    i := 1
    for i < n + 1 {
        if n % i == 0 {
            k--
            if k == 0 {
                return i
            }
        }
        i++
    }
    return -1
}
